network/src/protocols/discovery/mod.rs

```diff
@@ -163,8 +163,7 @@ where
     // list.
     async fn connect_to_seed_peers(&mut self) {
         debug!("Connecting to seed peers");
-        let self_peer_id =
-            PeerId::try_from(self.self_note.peer_id.clone()).expect("PeerId parsing failed");
+        let self_peer_id = self.self_peer_id;
```

Comment:
You should be able to remove this line

Reply:
```diff
@ network/src/protocols/discovery/mod.rs:169 @ where
    // list.
    async fn connect_to_seed_peers(&mut self) {
        debug!("Connecting to seed peers");
-       let self_peer_id = self.self_peer_id;
        for (peer_id, peer_info) in self
            .seed_peers
            .iter()
-           .filter(|(peer_id, _)| **peer_id != self_peer_id)
+           .filter(|(peer_id, _)| **peer_id != self.peer_id)
```

We get an error below if the line has been removed:

```
error[E0502]: cannot borrow `self.conn_mgr_reqs_tx` as mutable because it is also borrowed as immutable
   --> network/src/protocols/discovery/mod.rs:175:13
    |
170 |           for (peer_id, peer_info) in self
    |  _____________________________________-
171 | |             .seed_peers
172 | |             .iter()
173 | |             .filter(|(peer_id, _)| **peer_id != self.peer_id.clone())
    | |_____________________--------------______________----________________- immutable borrow later used here
    |                       |                           |
    |                       |                           first borrow occurs due to use of `self` in closure
    |                       immutable borrow occurs here
174 |           {
175 | /             self.conn_mgr_reqs_tx
176 | |                 .send(ConnectivityRequest::UpdateAddresses(
177 | |                     *peer_id,
178 | |                     peer_info
...   |
183 | |                         .collect(),
184 | |                 ))
    | |__________________^ mutable borrow occurs here
```

Reply:
Ahh ok, we'll keep it then.